In [0]:
from sklearn.datasets import load_wine
import pandas as pd

In [0]:
wine = load_wine()
dir(wine)

['DESCR', 'data', 'feature_names', 'frame', 'target', 'target_names']

In [0]:
df = pd.DataFrame(wine.data, columns=wine.feature_names)
df["target"] = wine.target
df.head()

alcohol 
 malic_acid 
 ash 
 alcalinity_of_ash 
 magnesium 
 total_phenols 
 flavanoids 
 nonflavanoid_phenols 
 proanthocyanins 
 color_intensity 
 hue 
 od280/od315_of_diluted_wines 
 proline 
 target 
 
 
 
 
 0 
 14.23 
 1.71 
 2.43 
 15.6 
 127.0 
 2.80 
 3.06 
 0.28 
 2.29 
 5.64 
 1.04 
 3.92 
 1065.0 
 0 
 
 
 1 
 13.20 
 1.78 
 2.14 
 11.2 
 100.0 
 2.65 
 2.76 
 0.26 
 1.28 
 4.38 
 1.05 
 3.40 
 1050.0 
 0 
 
 
 2 
 13.16 
 2.36 
 2.67 
 18.6 
 101.0 
 2.80 
 3.24 
 0.30 
 2.81 
 5.68 
 1.03 
 3.17 
 1185.0 
 0 
 
 
 3 
 14.37 
 1.95 
 2.50 
 16.8 
 113.0 
 3.85 
 3.49 
 0.24 
 2.18 
 7.80 
 0.86 
 3.45 
 1480.0 
 0 
 
 
 4 
 13.24 
 2.59 
 2.87 
 21.0 
 118.0 
 2.80 
 2.69 
 0.39 
 1.82 
 4.32 
 1.04 
 2.93 
 735.0 
 0

In [0]:
x = df.drop(['target'], axis='columns')
y = df.target

In [0]:
from sklearn.model_selection import train_test_split
x_train, x_test, y_train, y_test = train_test_split(x, y, train_size=0.7)

**Using naive bayes guassian Classifier**

In [0]:
from sklearn.naive_bayes import GaussianNB

In [0]:
gNB = GaussianNB()
gNB.fit(x_train, y_train)

GaussianNB()

In [0]:
gNB.score(x_test, y_test)

0.9814814814814815

**Using naive bayes multinominal Classifier**

In [0]:
from sklearn.naive_bayes import MultinomialNB

In [0]:
mNB = MultinomialNB()
mNB.fit(x_train, y_train)

MultinomialNB()

In [0]:

mNB.score(x_test, y_test)

0.8333333333333334

As, Guassien classifier is having a score of around 98% as compared to multinominal having score of around 83%.

**So, guassien classier have performed better!**